src/loss.py

```diff
@@ -1,2 +1,32 @@
 from abc import ABC
+import numpy as np
 
+
+class Purity_function(ABC):
+    def __init__(self):
+        pass
+
+    def purity_value(self, label_values: np.array, mask) -> float:
+        pass
+
+
+class Information_gaing(Purity_function):
+    super().__init__()
+
+
+
+    def _entropy(self, label_values: np.array) -> float:
+        _, uniqueClassesCounts = np.unique(label_values, return_counts=True)
+        probabilities = uniqueClassesCounts / uniqueClassesCounts.sum()
+        return sum(probabilities * -np.log2(probabilities))
+
+    def purity_value(self, labels_values: np.array, mask) -> float:
```

Comment:
type hint missing for mask

Reply:
from a first look, I think the type should be :
```suggestion
    def purity_value(self, labels_values: np.ndarray, mask: np.ndarray) -> float:
```

(FYI, the type hint for a numpy array is np.ndarray)